### PR TITLE
tka: move disablement logic out-of-band from AUMs

### PR DESCRIPTION
**Pull request by twitchyliquid64**
It doesn't make a ton of sense for disablement to be communicated as an AUM, because
any failure in the AUM or chain mechanism will mean disablement wont function.

Instead, tracking of the disablement secrets remains inside the state machine, but
actual disablement and communication of the disablement secret is done by the caller.

### CODE REVIEW

File: tka/aum.go [hunk 1]
```diff
@@ -189,23 +185,16 @@ func (a *AUM) StaticValidate() error {
...
 			return errors.New("Checkpoint AUMs may only specify State")
 		}
-	case AUMDisableNL:
-		if len(a.DisablementSecret) == 0 {
-			return errors.New("DisableNL AUMs must specify a disablement secret")
-		}
-		if a.KeyID != nil || a.State != nil || a.Key != nil || a.Votes != nil || a.Meta != nil {
-			return errors.New("DisableNL AUMs may only specify a disablement secret")
-		}
 	}
```
Comment (sailorfrag) [inline tka/aum.go:198]: reject unknown/unused message kinds?

File: tka/aum.go [hunk 2]
```diff
@@ -59,10 +59,9 @@ const (
 	//
 	// Only the KeyID optional field may be set.
 	AUMRemoveKey
-	// A DisableNL AUM describes the disablement of TKA.
-	//
-	// Only the DisablementSecret optional field may be set.
-	AUMDisableNL
+	// Formerly DisableNL, This AUM type is unused but kept to
+	// maintain ordering.
+	AUMUnused1
```
Comment (sailorfrag) [inline tka/aum.go:64]: I feel like maybe either:
- The test inputs that use this ID should be kept around to make sure that existing data continues to process without making things explode, or
- Removing the ID altogether makes sense, and we assume that old data using the old IDs will never be seen.

Right now it seems like if `AUMUnused1` is found, that's not an error, but nothing generates it anymore and it's ignored if seen.

**Review by sailorfrag — APPROVED** [review]

File: tka/aum.go (hunk 1 above)
Comment (twitchyliquid64) [reply tka/aum.go:196]: Done.

File: tka/aum.go (hunk 2 above)
Comment (twitchyliquid64) [reply tka/aum.go]: Ive gone ahead and removed it altogether, given theres nothing relying on this in the field yet. That did necessitate updating a bunch of tests, but no biggie.

**Review by sailorfrag — APPROVED** [review]

**Pull request merged**